### ISSUE TITLE
[BUG] App index error with grouped datasets

### ISSUE THREAD

**Issue by clane9** [voxel51/fiftyone#3307]
### System information

-   **OS Platform and Distribution** (e.g., Linux Ubuntu 16.04): macOS 12.6.5
-   **Python version** (`python --version`): 3.8.3
-   **FiftyOne version** (`fiftyone --version`): v0.21.4
-   **FiftyOne installed from** (pip or source): pip

### Describe the problem

I'm working with a grouped dataset where each group contains views from one of two non-overlapping sets. E.g., groups of type A have views a, b whereas groups of type B have views c, d, e. The dataset builds and loads without error. I'm able to look at samples from all of the views in the tile window. But I'm getting an error when I try to open a detailed window for a sample from group type B: "Error: sample with index 0 not found".

![image](https://github.com/voxel51/fiftyone/assets/11963692/60f2356e-02ba-42cf-985e-9eea5a8a4572)

### Code to reproduce issue

```python
from pathlib import Path

import fiftyone as fo
import numpy as np
from PIL import Image

def dummy_image(path, color):
    img = np.ones((64, 64, 3)) * color
    img = Image.fromarray(img.astype("uint8"))
    img.save(path)

dataset = fo.Dataset("group_test")
dataset.add_group_field("group")

img_dir = Path("group_test/images")
img_dir.mkdir(exist_ok=True, parents=True)

# Two sets of views. Each group contains views from only one split.
splits = {
    "A": {
        "a": (255, 0, 0),
        "b": (0, 255, 0),
    },
    "B": {
        "c": (0, 128, 128),
        "d": (128, 128, 0),
    }
}

samples = []
for ii in range(10):
    # If instead the group includes all splits/views, no error
    # group = fo.Group()
    for split, views in splits.items():
        group = fo.Group()
        for view, color in views.items():
            print(split, view)
            img_path = img_dir / f"{ii}_{split}_{view}.png"
            dummy_image(img_path, color)

            sample = fo.Sample(img_path, group=group.element(f"{split}/{view}"))
            samples.append(sample)

dataset.add_samples(samples)

ses = fo.launch_app(dataset)
ses.wait()
```

### What areas of FiftyOne does this bug affect?

-   [x] `App`: FiftyOne application issue
-   [ ] `Core`: Core Python library issue
-   [ ] `Server`: FiftyOne server issue

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently
-   [x] Yes. I would be willing to contribute a fix for this bug with guidance from the FiftyOne community
-   [ ] No. I cannot contribute a bug fix at this time

**Comment by clane9** [comment]
I tried to reproduce on a few other versions of fiftyone. On v0.21.0 I got a different error: "TypeError: Cannot read properties of null (reading 'sample')"

![image](https://github.com/voxel51/fiftyone/assets/11963692/8abd4729-d8f2-4949-b553-5141894027ca)

On v0.20.1 there was no error, the example worked fine.

**Comment by benjaminpkane** [comment]
Resolved in v0.21.6! Thanks @clane9
